Lib/test/test_sys.py

```diff
@@ -2152,5 +2153,31 @@ def test_remote_exec_invalid_script_path(self):
         with self.assertRaises(OSError):
             sys.remote_exec(os.getpid(), "invalid_script_path")
 
+    def test_remote_exec_in_process_without_debug_fails_envvar(self):
+        """Test remote exec in a process without remote debugging enabled"""
+        script = os_helper.TESTFN + '_remote.py'
+        self.addCleanup(os_helper.unlink, script)
+        with open(script, 'w') as f:
+            f.write('print("Remote script executed successfully!")')
+        env = os.environ.copy()
+        env['PYTHON_DISABLE_REMOTE_DEBUG'] = '1'
+
+        _, out, err = assert_python_failure('-c', f'import os, sys; sys.remote_exec(os.getpid(), "{script}")', **env)
```

Comment:
Random thought - should we special-case (or disallow) calling `remote_exec` on ourselves? It seems like it could be useful, but there's probably overheads involved that could be skipped. Alternatively, just saying "it's not for this" and blocking it also seems fine to me, though it'd complicate the tests.

Reply:
Nah, it actually requires no work whatsoever on our part, which is very cool! And we use it all the time for testing and developing since having to attach debuggers to two or even 3 processes is much much harder. 

I am happy to keep it unless you feel strongly